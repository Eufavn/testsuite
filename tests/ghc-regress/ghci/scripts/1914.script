:set -v1
System.IO.writeFile "A.hs" "module A where { import B; }"
System.IO.writeFile "B.hs" "module B where { type T = Int; }"
-- hack: avoid the need for sleep by using specific timestamps:
:! touch -t 01010000 A.hs
:! touch -t 01010000 B.hs
:load A
System.IO.writeFile "A.hs" "module A where { import B; oops }"
:! touch -t 01010001 A.hs
:reload
-- B should still be loaded
:module *B
System.IO.writeFile "A.hs" "module A where { import B; }"
:! touch -t 01010002 A.hs
:reload
-- should not have to reload B
