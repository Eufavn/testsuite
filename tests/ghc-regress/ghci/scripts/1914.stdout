Ok, modules loaded: A, B.
Failed, modules loaded: B.
Ok, modules loaded: A, B.
